keyboards/meletrix/zoom65_lite/keymaps/vial/keymap.c

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2021 meletrix
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
+    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [1] = { ENCODER_CCW_CW(KC_MS_WH_DOWN, KC_MS_WH_UP) },
+    [2] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS) },
+};
+#endif
+
+#ifdef VIA_ENABLE
+static uint8_t encoder_state[ENCODERS] = {0};
+static keypos_t encoder_cw[ENCODERS] = ENCODERS_CW_KEY;
+static keypos_t encoder_ccw[ENCODERS] = ENCODERS_CCW_KEY;
+
+void encoder_action_unregister(void)
+{
+    for (int index = 0; index < ENCODERS; ++index)
+    {
+        if (encoder_state[index])
+        {
+            keyevent_t encoder_event = (keyevent_t){
+                .key = encoder_state[index] >> 1 ? encoder_cw[index] : encoder_ccw[index],
+                .pressed = false,
+                .time = (timer_read() | 1)};
+            encoder_state[index] = 0;
+            action_exec(encoder_event);
+        }
+    }
+}
+void encoder_action_register(uint8_t index, bool clockwise)
+{
+    keyevent_t encoder_event = (keyevent_t){
+        .key = clockwise ? encoder_cw[index] : encoder_ccw[index],
+        .pressed = true,
+        .time = (timer_read() | 1)};
+    encoder_state[index] = (clockwise ^ 1) | (clockwise << 1);
+    action_exec(encoder_event);
+}
+
+void matrix_scan_kb(void)
+{
+    encoder_action_unregister();
+    matrix_scan_user();
+}
+
+bool encoder_update_kb(uint8_t index, bool clockwise)
+{
+    encoder_action_register(index, clockwise);
+    return true;
+};
+#endif
```

Comment:
What's that for?

Reply:
I am not the author of the functions, they were already presents in the VIA keymap for the Zoom65 Lite included in the upstream QMK repository and they are also present in the qmk-vial fork (I used those files as starting point to create this Vial keymap): https://github.com/vial-kb/vial-qmk/blob/vial/keyboards/meletrix/zoom65_lite/keymaps/via/keymap.c

I am not an expert but they look like handler functions for the encoder events, not sure if provided by Meletrix folks themselves. I just know that the handlers plus my Vial-specific additions work and properly support for the encoder.

Reply:
You should try deleting that part and check if it still works

Reply:
I commented the functions and the code still compiles (and the firmware works). I just pushed the change.